services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/service/MongoDbBasedRegistrationService.java

```diff
@@ -333,6 +354,53 @@ private RegistrationResult getRegistrationResult(final String deviceId, final Js
                 });
     }
 
+    private Future<OperationResult<List<DeviceWithId>>> processSearchDevices(
```

Comment:
is there any value in creating this private method with exactly the same signature as *searchDevices* instead of simply implementing *searchDevices* directly?

Reply:
As the code is already compact, IMHO there is no benefit or drawback. I just followed the same style as of other methods in that class.